### ISSUE TITLE
Can't view published flexdashboard as anonymous user

### ISSUE THREAD

**Issue by gordonwoodhull** [att/rcloud.flexdashboard#12]
_From @Prateek032 on November 28, 2016 10:30_

When anonymous user opens the published notebook in `flexdashboard.html`, it fails to display the contents.

Flexdashboard notebook contents are visible only when the user is **logged in** 

_Copied from original issue: att/rcloud#2366_

**Comment by gordonwoodhull** [comment]
Good work as always, @Prateek032! Too bad we didn't get you properly set up before the release, as this is a pretty bad bug. But since this is an independent package, we should be able to fix it without patching RCloud core.

Moving to att/rcloud.flexdashboard.

**Comment by gordonwoodhull** [comment]
Error displayed in browser console:

```
bluebird.js:908 Unhandled rejection TypeError: rcloud._ocaps.load_module_package is not a function
    at http://135.197.192.220:8080/shared.R/rcloud.flexdashboard/js/rcloud-flexdashboard.js:48:31
    at tryCatcher (http://135.197.192.220:8080/lib/js/bluebird.js:4409:31)
    at Promise._settlePromiseFromHandler (http://135.197.192.220:8080/lib/js/bluebird.js:2560:31)
    at Promise._settlePromiseAt (http://135.197.192.220:8080/lib/js/bluebird.js:2636:18)
    at Promise._settlePromises (http://135.197.192.220:8080/lib/js/bluebird.js:2752:14)
    at Async._drainQueue (http://135.197.192.220:8080/lib/js/bluebird.js:131:16)
    at Async._drainQueues (http://135.197.192.220:8080/lib/js/bluebird.js:141:10)
    at MutationObserver.Async.drainQueues (http://135.197.192.220:8080/lib/js/bluebird.js:66:14)
```

**Comment by gordonwoodhull** [comment]
We need to open up an ocap on RCloud for this - see att/rcloud#857. It would be safer to open up something more limited, rather than allowing anonymous users the ability to load arbitrary packages. We'll figure out something for RCloud 1.8.

**Comment by gordonwoodhull** [comment]
I'm taking this, as it has to do with weird ocap edge cases and may need to be fixed on the RCloud side.

**Comment by gordonwoodhull** [comment]
This was actually quite simple to fix, once I was able to test it properly. 

Since we can rely on `rcloud.flexdashboard` to be loaded via `rcloud.alluser.addons`, the OCAPs will already be available as long as the session specifies that it wants addons. 

Previously it was running in `mode: "client"` which did not load extensions, but I changed it to `mode: "addons"` and that worked. 

Note that `mode: "IDE"` did **not** work because that has two sessions, and I was getting bizarre RCS-not-initialized errors which I still don't understand. Anyway we don't want compute/control sessions, just a single compute session with addons running.

This means we can leave the `load_module_package` OCAP unavailable for anonymous sessions, which is probably safer.
